cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -200,7 +207,7 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 		var now time.Time
 		for now = timeNow(); now.Sub(start) < defaultMountUnitWaitTimeout; now = timeNow() {
 			mounted, err := osutilIsMounted(where)
-			if mounted {
+			if mounted != opts.Umount {
```

Comment:
nitpick: I think this would slightly more readable as:

`mounted == !opts.Unmount`